public/js/components/Editor.js

```diff
@@ -58,12 +58,18 @@ const Editor = React.createClass({
       return b.getIn(["location", "line"]) === line + 1;
     });
 
-    const applyBp = bp ? this.props.removeBreakpoint : this.props.addBreakpoint;
-    applyBp({
-      sourceId: this.props.selectedSource.get("id"),
-      line: line + 1,
-      snippet: cm.lineInfo(line).text.trim()
-    });
+    if (bp) {
+      this.props.removeBreakpoint({
+        sourceId: this.props.selectedSource.get("id"),
+        line: line + 1
+      });
+    } else {
+      this.props.addBreakpoint(
+        { sourceId: this.props.selectedSource.get("id"),
+          line: line + 1 },
+        { getTextForLine: l => cm.getLine(l - 1).trim() }
```

Comment:
Unfortunately, we need to get the text if a breakpoint slides. We only know if it slides once the protocol request completes, at which point we need to query the editor again, but we have no way to do that. A function allows us to compute the new value if it slides.